service/src/main/java/org/sunbird/service/user/UserProfileReadService.java

```diff
@@ -145,6 +145,37 @@ public Response getUserProfileData(Request actorMessage) {
     result.put(JsonKey.IDENTIFIER, userId);
 
     mapUserRoles(result);
+    if(result.containsKey(JsonKey.ROLES) && CollectionUtils.isNotEmpty((Collection) result.get(JsonKey.ROLES))){
+      List<String> mentoringRoles = new ArrayList<>();
+      List<String> roles = new ArrayList<>();
+      if (result.get(JsonKey.ROLES) instanceof List) {
+        List<?> list = (List<?>) result.get(JsonKey.ROLES);
+
+        if (!list.isEmpty() && list.get(0) instanceof HashMap) {
+          for (Object element : list) {
+            HashMap<?, ?> hashmap = (HashMap<?, ?>) element;
+            if (hashmap.containsKey(JsonKey.ROLE) && hashmap.get(JsonKey.ROLE) instanceof String) {
+              roles.add((String) hashmap.get(JsonKey.ROLE));
+            }
+          }
+        } else if (!list.isEmpty() && list.get(0) instanceof String) {
```

Comment:
Please handle null pointer and try to use collectionUtils as it handle the proper null pointer